apps/storybook/src/overlays/Spotlight.stories.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  Highlight,
+  Spotlight,
+  SpotlightContent,
+  SpotlightEmpty,
+  SpotlightInput,
+  SpotlightItem,
+  SpotlightLabel,
+  SpotlightList,
+  SpotlightTrigger,
+} from "@optiaxiom/react/unstable";
+import { Fragment, useState } from "react";
+
+export default {
+  args: {
+    defaultOpen: true,
+  },
+  component: Spotlight,
+  parameters: {
+    useOverlayDecorator: true,
+  },
+} as Meta<typeof Spotlight>;
+
+type Story<T> = StoryObj<typeof Spotlight<T>>;
+
+const pages = [
+  {
+    category: "Layout",
+    description: "Set the element's `gap` CSS property",
+    title: "Gap",
+  },
+  {
+    category: "Layout",
+    description: "Set the element's margin on all sides",
+    title: "Margin",
+  },
+  {
+    category: "Typography",
+    description: "Set the element's font family",
+    title: "Font Family",
+  },
+  {
+    category: "Typography",
+    description:
+      "Set the element's `font-size` and `line-height` CSS properties",
+    title: "Font Size",
+  },
+  {
+    category: "Typography",
+    description: "Set the element's text color",
+    title: "Text Color",
+  },
+  {
+    category: "Sizing",
+    description: "Set the element's height",
+    title: "Height",
+  },
+  {
+    category: "Sizing",
+    description: "Set the element's width",
+    title: "Width",
+  },
+];
+
+export const Basic: Story<(typeof pages)[number]> = {
+  render: function Basic(args) {
+    const [open, setOpen] = useState(args.defaultOpen);
+    const [inputValue, setInputValueState] = useState("");
+    const [items, setItems] = useState(pages);
+
+    const setInputValue = (inputValue: string) => {
+      setInputValueState(inputValue);
+      setItems(
+        pages.filter(
+          (page) =>
+            !inputValue ||
+            getAllPhrasesMatchingRegex(
+              inputValue
+                .split(" ")
+                .map((q) => q.trim())
+                .filter(Boolean),
+            ).test(page.description + "\n" + page.title),
+        ),
+      );
+    };
+
+    const categories = new Set();
+    const shouldShowCategory = (category: string) => {
+      const flag = categories.has(category);
+      categories.add(category);
+      return !flag;
+    };
+
+    return (
+      <Spotlight
+        {...args}
+        inputValue={inputValue}
+        items={items}
+        onInputValueChange={setInputValue}
+        onItemSelect={(value) => {
+          toaster.create(
+            <Toast colorScheme="success">
+              <ToastTitle>
+                Selected <strong>{value.title}</strong>
+              </ToastTitle>
+            </Toast>,
+          );
+        }}
+        onOpenChange={setOpen}
+        open={open}
+      >
+        <SpotlightTrigger
+          addonAfter={
+            <Kbd keys="command" variant="subtle">
+              K
+            </Kbd>
+          }
+          w="240"
+        >
+          Quick search...
+        </SpotlightTrigger>
+
+        <SpotlightContent>
```

Comment:
Bhai, why aren't we passing the `SpotlightInput` in the addOnBefore prop of `Spotlight`

Reply:
So addons are slots that let you add/replace things down the chain

A good example is https://github.com/optimizely-axiom/optiaxiom/blob/5f763137e5fbbf44f391571e0040f699ad92bae6/packages/react/src/command-checkbox-item/CommandCheckboxItem.tsx#L20-L29

where we’re adding a checkbox before any other thing in the addon

another example is inputs where we want the addon to go inside the container but next to the actual inputs

and another example is AlertDialogTitle which allows consumers to actually replace the alert icon with something they’d like

And in some cases addons have special styles depending on if they’re placed in the start or end

I suppose we could’ve instead broken these components down into their individual parts and made them really modular - like:

```tsx
<Input>
  <InputAddon />
  <InputControl />
</Input>
```

but that makes the api a bit cumbersome for relatively simple things - hence why we have this addon approach

But in this case with Spotlight we’re just doing pure composition - and there’s no need to create slots.

now that you bring it up - I think we should actually remove the addons from Sidenav as well - since that doesn’t really need slots.

Reply:
Yes bhai. I kind of related this part with Sidenav and as such had this query. Thank you for the descriptive reply!